article/_posts/2022-01-14-CAVAT.md

```diff
@@ -0,0 +1,91 @@
+optimizing a non-differentiable loss function with the REINFORCE algorithm.
+
+Authors start by listing a number of methods that attempt to add constraints to neural network training such as size 
+constraints and centroid positions. These constraints require differentiable loss functions which limit their use. 
+Authors would like to exploit the information contained in anatomical shape priors like in Oktay et al., 2017 (ACNN), Zotti
+et al., 2018 (Prior-aware segmentation) and Painchaud et al. 2020. 
+
+Authors explore the aspect of semi-supervised learning as training segmentation networks to learn valid shapes is a 
+difficult task when little labeled data is available. Authors, therefore, propose a method base on Virtual Adversarial 
+Training (VAT) and REINFORCE to optimize a non-differentiable local connectivity constraint. 
+
+# Method 
+
+## Note on VAT 
+The goal of the VAT paper (Virtual Adversarial Training: A Regularization Method for Supervised and Semi-Supervised 
+Learning) is to propose a method for semi-supervised learning based on regularization induced by adversarial 
+perturbations. The loss is given by: 
+
+![](/article/images/cavat/vatloss.JPG)
+
+Here, an adversarial perturbation is generated using gradient descent to maximize the divergence in predictions of the 
+network.
+
+## Training
+![](/article/images/cavat/fig1.JPG)
+
+Training the proposed method is done on a labeled set $$\matcal{S}$$ and an un-labled set $$mathcal{U}$$. The network is 
+trained using the following loss function.
+
+![](/article/images/cavat/equation1.JPG)
+
+The supervised loss $$\mathcal{L}_{sup}$$. The Context-aware VAT loss term is composed of two losses. 
+
+![](/article/images/cavat/equation3.JPG)
+
+The first loss is the local distribution smoothness (LSD), which is directly taken from VAT. This loss regularized the 
+network by making it more robust to examples that violate the virtual adversarial direction.  
+
+![](/article/images/cavat/equation4.JPG)
+
+The reinforced constraint loss is given by 
+
+![](/article/images/cavat/equation7.JPG)
+
+where $$J$$ is the per-pixel reward and $$p_u = f(x_u+r_r;\theta)$$ is the probability output of the network. 
+
+
+Looking back at equation (3), we can note here that the adversarial perturbation $$r_U$$ is maximized on both terms of 
+the CAVAT loss which increases the number of samples that violate the constraint. 
+
+## Local connectivity constraint
+
+The constraint optimized in this paper is the local connectivity constraint. Connectivity if a region G means that there 
+exists a path between every pair of pixels in G such that every pixel in that path is also in G. As global connectivity
+is hard to achieve at the beginning of training, authors propose to optimize local connectivity at local patches.
+
+Given a discreate prediction $$\hat{y}$$. The others apply a $$1_{ l\times l}$$ convolution to find the pixel with the 
```

Comment:
given a discreate prediction $$\hat{y}$$. -> Unfinished sentence.

Reply:
Fixed in bdcc473